### PR TITLE
Laboratory work #7, Irina Kurnakova - 20FPL1

### PR DESCRIPTION
**Pull request by IrinaKurnakova**

### CODE REVIEW

File: lab_7_llm/main.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 from pathlib import Path
 from typing import Iterable, Iterator, Sequence
 
+
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Delete

**Comment by SalnikovIgor** [comment]
Necessary implementation of loading your dataset.

**Comment by SalnikovIgor** [comment]
It's time to write the loading of your dataset.

**Comment by SalnikovIgor** [comment]
And again: it's time to write the loading of your dataset.

File: lab_7_llm/main.py [hunk 2]
```diff
@@ -41,6 +42,11 @@ def obtain(self) -> None:
         Raises:
             TypeError: In case of downloaded dataset is not pd.DataFrame
         """
+        dataset_df = load_dataset(self._hf_name, split='test').to_pandas()
+        self._raw_data = dataset_df
```
Comment (SalnikovIgor) [inline lab_7_llm/main.py]: Two operations can be reduced into one

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
Fix comments and implement model analysis

**Comment by IrinaKurnakova** [comment]
I don't know what's wrong with tests(

File: lab_7_llm/start.py [hunk 3]
```diff
@@ -2,15 +2,40 @@
...
+
+    dataset = TaskDataset(preprocessor.data.head(100))
+
+    pipeline = LLMPipeline(settings['parameters']['model'], dataset,
+                           max_length=512, batch_size=1, device='cpu')
+
+    model_analysis = pipeline.analyze_model()
+    print(model_analysis)
+
+    sample = dataset.__getitem__(0)
```
Comment (SalnikovIgor) [inline lab_7_llm/start.py]: You shouldn't take one element using a dunder methods. If you can't use` next ... iter` use `for ... in ...` once

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
When do you plan to implement a service for a score of 10?

**Review by SalnikovIgor — CHANGES_REQUESTED** [review]
It's time to move on to task number 10

**Pull request closed without merging**